dev/com.ibm.ws.microprofile.metrics.3.0/src/com/ibm/ws/microprofile/metrics30/helper/Util30.java

```diff
@@ -0,0 +1,150 @@
+                                                                String metricName) throws NoSuchRegistryException, EmptyRegistryException, NoSuchMetricException {
+        MetricRegistry registry = getRegistry(registryName);
+        Map<String, Metadata> metricMetadataMap = new HashMap<String, Metadata>();
+        for (String name : registry.getNames()) {
+            metricMetadataMap.put(name, registry.getMetadata(name));
+        }
+
+        Map<String, Metadata> returnMap = new HashMap<String, Metadata>();
+
+        if (metricMetadataMap.isEmpty()) {
+            throw new EmptyRegistryException();
+        } else if (!(metricMetadataMap.containsKey(metricName))) {
+            throw new NoSuchMetricException();
+        } else {
+            returnMap.put(metricName, metricMetadataMap.get(metricName));
+        }
+        return returnMap;
+    }
+
+    public static Map<String, Number> getTimerNumbers(Timer timer, String tags, double conversionFactor) {
+        Map<String, Number> results = new HashMap<String, Number>();
+        results.put(Constants.COUNT + tags, timer.getCount());
+        results.put("elapsedTime" + tags, timer.getElapsedTime().toNanos());
+        results.put(Constants.MEAN_RATE + tags, timer.getMeanRate());
+        results.put(Constants.ONE_MINUTE_RATE + tags, timer.getOneMinuteRate());
+        results.put(Constants.FIVE_MINUTE_RATE + tags, timer.getFiveMinuteRate());
+        results.put(Constants.FIFTEEN_MINUTE_RATE + tags, timer.getFifteenMinuteRate());
+
+        results.put(Constants.MAX + tags, timer.getSnapshot().getMax() / conversionFactor);
+        results.put(Constants.MEAN + tags, timer.getSnapshot().getMean() / conversionFactor);
+        results.put(Constants.MIN + tags, timer.getSnapshot().getMin() / conversionFactor);
+
+        results.put(Constants.STD_DEV + tags, timer.getSnapshot().getStdDev() / conversionFactor);
+
+        results.put(Constants.MEDIAN + tags, timer.getSnapshot().getMedian() / conversionFactor);
+        results.put(Constants.PERCENTILE_75TH + tags, timer.getSnapshot().get75thPercentile() / conversionFactor);
+        results.put(Constants.PERCENTILE_95TH + tags, timer.getSnapshot().get95thPercentile() / conversionFactor);
+        results.put(Constants.PERCENTILE_98TH + tags, timer.getSnapshot().get98thPercentile() / conversionFactor);
+        results.put(Constants.PERCENTILE_99TH + tags, timer.getSnapshot().get99thPercentile() / conversionFactor);
+        results.put(Constants.PERCENTILE_999TH + tags, timer.getSnapshot().get999thPercentile() / conversionFactor);
+
+        return results;
+    }
+
+    /*
+     * To facilitate changes into the MicroProfile Specification in 3.0 we need to return a value that is an
+     * Object because the maxTimeDuration or minTimeDuration could be a NaN object.
+     */
+    public static Map<String, Object> getSimpleTimerNumbersAsObjects(SimpleTimer simpleTimer, String tags, double conversionFactor) {
```

Comment:
Needed a new method that returns a map with Object values. This is due to the possibility of a null value being returned.